app/controllers/concerns/foreman/controller/registration_commands.rb

```diff
@@ -27,15 +30,33 @@ def registration_url(proxy = nil)
     "#{url}/register"
   end
 
+  def jwt_expiration_param
+    param = registration_params['jwt_expiration'] || 4
+    @jwt_expiration_param ||= begin
+      if param == 'unlimited'
+        0
+      else
+        param.to_i
```

Comment:
This implementation will not fail validation of strings other than `unlimited`.

Since you have to accept decimal values, you can still have it validated:

```suggestion
        Float(param).to_i
```
This will accept `11.5`, but will fail for unknown words.

Reply:
for the input let's say 11.5, the output will be the same for `param.to_i` or `Float(param).to_i`
![Screenshot from 2024-01-04 17-13-18](https://github.com/theforeman/foreman/assets/115526987/5f604097-03f7-421e-9d75-d52ac035c0fe)